qa-ovh-maintenance/update_static_slaves.groovy

```diff
@@ -1,48 +1,92 @@
-/*
- * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- *
- * Contributors:
- * - atimic
- * - jcarsique
- *
- * QA OVH
- */
 
 import hudson.model.Cause;
 import jenkins.model.Jenkins;
 
 def update_static_slaves(boolean doConfirm=false) {
+
+  // Get All static slaves
   def staticSlaves = [];
+  def offlineIdleSlaves = [];
+  def onlineBusySlaves = [];
+  def availableSlaves = [];
   for (slave in Jenkins.instance.getNodes()) { // iterate on all slaves
-    for (label in slave.getLabelString().split()) { // look for a known "static" label
-      if ("STATIC".equalsIgnoreCase(label)) {
-          if (slave.toComputer().isOnline() && slave.toComputer().isIdle()) {
-              staticSlaves.add(slave.getDisplayName());
-          } else {
-              println 'Ignore unavailable slave ' + slave.getDisplayName()
-          }
-          break
+    if (slave.toComputer().getConnectTime() > 0) {
+      for (label in slave.getLabelString().split()) { // look for a known "static" label
+        if ("STATIC".equalsIgnoreCase(label)) {
+          availableSlaves.add(slave.getDisplayName());
+          staticSlaves.add(slave);
+          println("adding " + slave.getDisplayName() + " to availableSlaves");
+        }
       }
     }
   }
+  // Check if those slaves are outdated | if yes, write slave name to result.txt
+  sh """ #!bin/bash -x
+  rm -f result.txt
+  for i in 1 2 3; do
+    cd $WORKSPACE/qa-ovh-maintenance/qa-ovh0"\${i}"/
+    for slave in ${availableSlaves}; do
+      slave=\${slave/[/} && slave=\${slave/]/} && slave=\${slave/,/}
+      echo "\$slave"
+      ssh jenkins@qa-ovh0"\${i}".nuxeo.com "bash -s -- \${slave}" < . ../common/check.sh
```

Comment:
` || true` or `set +e` rather than removing the whole error-on-exit behavior

Reply:
you can remove the remote ssh with a local call to 
`docker -H qa-ovh01.nuxeo.com:4243 ps -f name=itslave01 --format '{{.ID}}'`
or 
`docker -H tcp://swarm-qa.nuxeo.org:4000 info`

Reply:
(alternate way could have been to retrieve the SH output in a Groovy variable for later treatment)